src/app/core/entity/entity-actions/entity-actions.service.spec.ts

```diff
@@ -0,0 +1,132 @@
+  let service: EntityActionsService;
+  let mockedEntityMapper: jasmine.SpyObj<EntityMapperService>;
+  let snackBarSpy: jasmine.SpyObj<MatSnackBar>;
+  let mockSnackBarRef: jasmine.SpyObj<MatSnackBarRef<TextOnlySnackBar>>;
+  let mockConfirmationDialog: jasmine.SpyObj<ConfirmationDialogService>;
+  let mockRouter;
+  let mockedEntityDeleteService: jasmine.SpyObj<EntityDeleteService>;
+
+  let primaryEntity: Entity;
+
+  beforeEach(() => {
+    primaryEntity = new Entity();
+
+    mockedEntityDeleteService = jasmine.createSpyObj(["deleteEntity"]);
+    mockedEntityDeleteService.deleteEntity.and.resolveTo([primaryEntity]);
+    mockedEntityMapper = jasmine.createSpyObj(["save", "saveAll"]);
+    snackBarSpy = jasmine.createSpyObj(["open"]);
+    mockSnackBarRef = jasmine.createSpyObj(["onAction", "afterDismissed"]);
+    mockSnackBarRef.onAction.and.returnValue(of());
+    mockConfirmationDialog = jasmine.createSpyObj(["getConfirmation"]);
+    mockConfirmationDialog.getConfirmation.and.resolveTo(true);
+    snackBarSpy.open.and.returnValue(mockSnackBarRef);
+
+    TestBed.configureTestingModule({
+      imports: [CoreTestingModule],
+      providers: [
+        EntityActionsService,
+        { provide: EntityDeleteService, useValue: mockedEntityDeleteService },
+        { provide: EntityAnonymizeService, useValue: null },
+        { provide: EntityMapperService, useValue: mockedEntityMapper },
+        { provide: MatSnackBar, useValue: snackBarSpy },
+        Router,
+        {
+          provide: ConfirmationDialogService,
+          useValue: mockConfirmationDialog,
+        },
+      ],
+    });
+    mockRouter = TestBed.inject(Router);
+    spyOn(mockRouter, "navigate");
+
+    service = TestBed.inject(EntityActionsService);
+  });
+
+  it("should return false when user cancels confirmation", async () => {
+    mockConfirmationDialog.getConfirmation.and.resolveTo(false);
+
+    const result = await service.delete(new Entity());
+
+    expect(result).toBe(false);
+    expect(snackBarSpy.open).not.toHaveBeenCalled();
+    expect(mockedEntityDeleteService.deleteEntity).not.toHaveBeenCalled();
+  });
+
+  it("should delete entity, show snackbar confirmation and navigate back", async () => {
+    // onAction is never called
+    mockSnackBarRef.onAction.and.returnValues(NEVER);
+    mockSnackBarRef.afterDismissed.and.returnValue(of(undefined));
+
+    const result = await service.delete(new Entity(), true);
+
+    expect(result).toBe(true);
+    expect(snackBarSpy.open).toHaveBeenCalled();
+    expect(mockedEntityDeleteService.deleteEntity).toHaveBeenCalled();
+    expect(mockRouter.navigate).toHaveBeenCalled();
+  });
+
+  it("should re-save all affected entities and navigate back to entity on undo", fakeAsync(() => {
+    const anotherAffectedEntity = new Entity();
+    mockedEntityDeleteService.deleteEntity.and.resolveTo([
+      primaryEntity,
+      anotherAffectedEntity,
+    ]);
+
+    // Mock a snackbar where 'undo' is immediately pressed
```

Comment:
Is 'immediate' relevant here?

Reply:
this is from the tests that had been present originally. Indeed, I would understand this also that it is not immediate but once "next" is fired.

Simplified/Corrected now.